test/components/PostList.test.js

```diff
@@ -78,13 +43,55 @@ const posts = [
   }
 ]
 
-describe('PostList', () => {
-  it('renders a mixed group of posts', () => {
-    const component = createElement(PostList, {posts}, {store})
-    const doc = cheerio.load(renderToString(component))
-    expect(doc('.post').length).to.equal(3)
-    expect(doc('.project-summary').length).to.equal(1)
-    expect(doc('.event-summary').length).to.equal(1)
-    expect(doc('.my-module').length).to.equal(1)
+describe('<PostList />', () => {
+  it('should shallow render', () => {
+    const wrapper = shallow(<PostList posts={posts} />)
+    expect(wrapper.find('ShowPost').length).to.equal(4)
+    expect(
+      wrapper.find('ShowPost').map((postWrapper) => postWrapper.dive().name())
```

Comment:
Right? With a little "dive" now and then shallow rendering can cover just about any case.